File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/startup/StartupActionsListMessage.scala





















// Generated on 05/03/2013 10:35:00
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait StartupActionsListMessage extends MessageDeserializer {
  val messageId = 1301

  protected trait Message extends BaseMessage with Serializable {
    val messageId = StartupActionsListMessage.messageId

    val actions: Seq[StartupActionAddObject.TypeImpl]

    def serialize(buf: Buffer) {
      buf.writeUShort(actions.length)
      for (entry <- actions) {
        entry.serialize(buf)
      }
    }

  }

  protected def create(actions: Seq[StartupActionAddObject.TypeImpl]): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val actions = {
      val builder = collection.immutable.Seq.newBuilder[StartupActionAddObject.TypeImpl]
      for (i <- (0 to buf.readUShort())) {
        builder += StartupActionAddObject.deserialize(buf)
      }
      builder.result()
    }

    create(actions)
  }

}

object StartupActionsListMessage extends StartupActionsListMessage {
  case class MessageImpl(actions: Seq[StartupActionAddObject.TypeImpl]) extends Message
  type Target = MessageImpl

  override def create(actions: Seq[StartupActionAddObject.TypeImpl]) = MessageImpl(actions)

  def apply(actions: Seq[StartupActionAddObject.TypeImpl]) = create(actions)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/StartupActionsListMessage.java


// Generated on 05/08/2013 19:37:58
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class StartupActionsListMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 1301;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public StartupActionAddObject[] actions;
    
    public StartupActionsListMessage() { }
    
    public StartupActionsListMessage(StartupActionAddObject[] actions) {
        this.actions = actions;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeUShort(actions.length);
        for (StartupActionAddObject entry : actions) {
            entry.serialize(buf);
        }
    }
    
    @Override
    public void deserialize(Buffer buf) {
        int limit = buf.readUShort();
        actions = new StartupActionAddObject[limit];
        for (int i = 0; i < limit; i++) {
            actions[i] = new StartupActionAddObject();
            actions[i].deserialize(buf);
        }
    }
    
}
